artemis-commons/src/main/java/org/apache/activemq/artemis/utils/CompositeAddress.java

```diff
@@ -41,12 +41,36 @@ public static SimpleString toFullyQualified(SimpleString address, SimpleString q
       return result;
    }
 
+   /**
+    * Checks whether a given string complies the fully qualified queue name (FQQN) standard or not.
+    *
+    * @param address The string to check for FQQN standard compliance.
+    * @return boolean
+    */
    public static boolean isFullyQualified(String address) {
-      return address == null ? false : address.contains(SEPARATOR);
+      if (address == null) {
+         return false;
+      }
+
+      // If address length is incapable of holding the separator and two more characters
+      if (address.length() < SEPARATOR.length() + 2) {
+         return false;
+      }
+
+      int index = address.indexOf(SEPARATOR, 1);
+
+      // If separator is not found, at the start or end of string then this is not a FQQN
+      return index != -1 && index != 0 && index != address.length() - 2;
```

Comment:
Middle check seems superfluous, i.e can index ever be 0 given the search starts at 1?

Reply:
You're right of course, just fixed it! I also changed the `address.length() - 2;` to `address.length() - SEPARATOR_LENGTH;` so that changing the separator does not affect the correct behaviour of the function!